src/sentry/sentry_apps/apps.py

```diff
@@ -185,30 +187,31 @@ def _update_events(self) -> None:
             self.sentry_app.events = expand_events(self.events)
 
     def _update_service_hooks(self) -> None:
-        hooks = hook_service.update_webhook_and_events(
-            organization_id=self.sentry_app.owner_id,
-            application_id=self.sentry_app.application_id,
+        if self.sentry_app.is_published:
+            # if it's a published integration, we need to do many updates so we have to do it in a task so we don't time out
+            # the client won't know it succeeds but there's not much we can do about that unfortunately
```

Comment:
Please add logs on failure, so at least we have a way of monitoring if something goes wrong

Reply:
@sentaur-athena are you talking about the task? Don't we automatically capture errors for every task?